### ISSUE TITLE
Requests Connection Reuse

### ISSUE THREAD

**Issue by jamesdreid** [p4r4n0y1ng/pyfmg#13]
Currently, the _post_request function uses the "requests.post" call to establish the connection to the FMG to make the API calls.  In this setup, it forces a new socket to be created for each call to the FMG and does not take advantage of the Requests library's builtin connection reuse functions.  Creating a socket variable by adding the following line at 172:

`self._socket = requests.session()`

And then changing line 277 from:

` response = requests.post(self._url, data=json.dumps(json_request), headers=headers, verify=self.verify_ssl,`

TO:

` response = self._socket.post(self._url, data=json.dumps(json_request), headers=headers, verify=self.verify_ssl,`

Will change this behavior and will allow the HTTPS socket to be reused for numerous call and reduce the load required to setup all the secure connections on the FMG when making numerous calls requesting data.

**Comment by p4r4n0y1ng** [comment]
I like this. Let me test it out and I'll just throw it in as a small commit and upgrade.

**Comment by jamesdreid** [comment]
I have not tested all of the functions with this change so I cannot confirm it does not break anything but I made a similar change in the FNDN Python lib last year and I have not encountered any issues yet.  I did run a simple performance test with this change with a script that would pull a list of devices from an ADOM and then run the Policy status API call (URL: /pm/config/adom/{{ADOM}}/_package/status/{{device}}/root) for every device (~180).  Using the Requests Post call, the script creates ~182 different sessions and take 53 seconds to complete.  With the session object, the script creates a single session and completes in 28 seconds so there is a significant speed increase.
